enhancements/local-storage/automatic-detection.md

```diff
@@ -64,40 +65,299 @@ and we can use that information to sort devices into storage classes based on th
+  - When the `discoveredTimeStamp` timestamp for each `LocalVolumeDisocveryResult`CR is greater than `discoveryStarted` timestamp, then update the `Phase` in the `LocalVolumeDiscovery` CR to `Discovered`.
+- Controller should update the `LocalVolumeDiscovery` `Phase` as `Error` if:
+  - There is some error in running the `DaemonSet`
+  - The`LocalVolumeDiscoveryResult` CR is not created for each valid nodes even after a fixed interval time.
+  - `discoveredTimeStamp` is not available in one of more `LocalVolumeDiscoveryResult` CR.
+
+#### `Auto Device Discovery` Cadence
+- Auto discovery of devices should be a continuous proccess to support on-going management of devices and to display the devices in an inventory list.
+- Auto discovery should continuously check for the list of available devices on a given node after every fixed interval of time to identify if a new device has been added or an existing device has been removed.
+- In case of any addtion or deletion of devices, the `LocalVolumeDiscoveryResult` CR should be updated to reflect the new set of devices on the node.
+- Continuous discovery can be triggered based following criteria:
+  - After every fixed interval of time.
+  - After any `udev` events on the block device (like `add` or `remove`)
+
+#### `Auto Device Discovery` Node Deletion
+- User updates the `LocalVolumeDiscovery` CR to remove node(s) for discovery:
+  - Daemonset would be updated with the new set of nodes, by the controller.
+  - This would automatically delete the daemonset pod running on the removed node(s).
+  - Controller would delete the `LocalVolumeDiscoveryResult` CR for the removed Node. (Can the `LocalVolumeDiscoveryResult` CR be garbage collected?)
+
+#### `Auto Device Discovery` Node Addition
+- User updates the `LocalVolumeDiscovery` CR to track new node(s) for discovery:
+  - DaemonSet would be updated with the new set of nodes, by the controller.
+  - This would automatically create a daemonset pod and start discovery on the newly added node(s).
+
+#### `Auto Device Discovery` CR Deletion
+- A delete request on the `LocalVolumeDiscovery` CR would trigger a graceful deletion with help of a finalizer and it would:
+  - Delete the `LocalVolumeDiscoveryResult` CR for each node. (Can the `LocalVolumeDiscoveryResult` CR be garbage collected?)
+  - Delete the `LocalVolumeDiscoveryResult` CRD.
+  - `Daemonset` gets garbage collected
+- In case of multiple `LocalVolumeDiscovery` CR, controller should:
+  - Analyze all the existing `LocalVolumeDiscover` CR and identify nodes on which the discovery `daemonset` are running.
+  - Identify the nodes that are exclusively available in the `LocalVolumeDiscovery` result being deleted.
+  - Update the `dameonset` to remove the nodes only available on `LocalVolumeDiscovery` being deleted.
+
+#### Multiple `Auto Device Discovery` CR
```

Comment:
Should we support multiple `LocalVolumeDiscovery` CR creation? @jsafrane @gnufied

Reply:
- We really only expect a single discovery CR, so supporting multiple discovery CRs is questionable IMO.
- If we did support multiple discover CRs, each CR should have its own daemonset defined. Supporting the same daemonset for multiple discovery CRs would be a challenge to combine the placement.

Reply:
@gnufied @jsafrane any thoughts on this ?

Reply:
Based on the latest discussion, we can have only one Local Volume Discovery CR for this release. Later, we can change this based on any new use cases.